src/Computation/LazySolver.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using Microsoft.Z3;
+using Symbolica.Expression;
+
+namespace Symbolica.Computation;
+
+internal sealed class LazySolver : ISolver
+{
+    private readonly IList<Action<ISolver>> _actions;
+    private ISolver? _solver;
+
+    public LazySolver()
+    {
+        _solver = null;
+        _actions = new List<Action<ISolver>>();
+    }
+
+    public Context Context => GetSolver().Context;
+
+    public void Dispose()
+    {
+        _solver?.Dispose();
+    }
+
+    public void Assert(IEnumerable<IValue> assertions)
+    {
+        _actions.Add(s => s.Assert(assertions));
```

Comment:
What about assertions that are added after the solver is created?

Reply:
Should be fine and follow the same path as any done before the solver is created. Unless I have messed up the implementation of course.

Reply:
Ah I see now. Everything goes through `GetSolver()` to get the solver and so the actions are always applied before the solver is returned. I thought it was just creation time laziness. 👍